Makefile

```diff
@@ -0,0 +1,266 @@
+ARCHITECTURES = amd64 arm64 arm ppc64le s390x
+
+.PHONY: ensure-version
+ensure-version:
+	node ./aio/scripts/version.mjs
+
+.PHONY: ensure-golangcilint
+ensure-golangcilint:
+ifndef GOLANGCILINT_BINARY
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOPATH)/bin $(GOLANGCILINT_VERSION)
+endif
+
+.PHONY: ensure-codegen
+ensure-codegen: ensure-go
+	go get -d k8s.io/code-generator@$(CODEGEN_VERSION)
+	chmod +x $(CODEGEN_BIN)
+
+.PHONY: ensure-fswatch
+ensure-fswatch:
+ifndef FSWATCH_BINARY
+	$(error "Cannot find fswatch binary")
+endif
+
+.PHONY: ensure-go
+ensure-go:
+ifndef GO_BINARY
+	$(error "Cannot find go binary")
+endif
+	@if [ $(GO_MAJOR_VERSION) -gt $(MIN_GO_MAJOR_VERSION) ]; then \
+		exit 0 ;\
+  elif [ $(GO_MAJOR_VERSION) -lt $(MIN_GO_MAJOR_VERSION) ]; then \
+		exit 1; \
+  elif [ $(GO_MINOR_VERSION) -lt $(MIN_GO_MINOR_VERSION) ] ; then \
+		exit 1; \
+  fi
+
+.PHONY: clean
+clean:
+	rm -rf .tmp
+
+.PHONY: build-backend
+build-backend: ensure-go
+	go build -ldflags "-X $(MAIN_PACKAGE)/client.Version=$(RELEASE_VERSION)" -gcflags="all=-N -l" -o $(SERVE_BINARY) $(MAIN_PACKAGE)
+
+.PHONY: build
+build: clean ensure-go
+	./aio/scripts/build.sh
+
+.PHONY: build-cross
+build-cross: clean ensure-go
+	./aio/scripts/build.sh -c
+
+.PHONY: serve-backend
+serve-backend: build-backend
+	cp i18n/locale_conf.json $(SERVE_DIRECTORY)/locale_conf.json
+	$(SERVE_BINARY) --kubeconfig=$(KUBECONFIG) \
+		--sidecar-host=$(SIDECAR_HOST) \
+		--system-banner=$(SYSTEM_BANNER) \
+		--system-banner-severity=$(SYSTEM_BANNER_SEVERITY) \
+		--token-ttl=$(TOKEN_TTL) \
+		--auto-generate-certificates=$(AUTO_GENERATE_CERTS) \
+		--enable-insecure-login=$(ENABLE_INSECURE_LOGIN) \
+		--enable-skip-login=$(ENABLE_SKIP_LOGIN) & echo $$! > $(SERVE_PID)
+
+.PHONY: kill-backend
+kill-backend:
+	kill `cat $(SERVE_PID)` || true
+	rm -rf $(SERVE_PID)
+
+.PHONY: restart-backend
+restart-backend: kill-backend serve-backend
+
+.PHONY: watch-backend
+watch-backend: ensure-fswatch restart-backend
+	fswatch -o -r -e '.*' -i '\.go$$'  . | xargs -n1 -I{} make restart-backend || make kill-backend
+
+.PHONY: prod-backend
+prod-backend: clean ensure-go
+	GOOS=linux go build -a -installsuffix cgo -ldflags "-X $(MAIN_PACKAGE)/client.Version=$(RELEASE_VERSION)" -o $(PROD_BINARY) $(MAIN_PACKAGE)
+
+.PHONY: prod-backend-cross
+prod-backend-cross: clean ensure-go
+	for ARCH in $(ARCHITECTURES) ; do \
+  	GOOS=linux GOARCH=$$ARCH go build -a -installsuffix cgo -ldflags "-X $(MAIN_PACKAGE)/client.Version=$(RELEASE_VERSION)" -o dist/$$ARCH/dashboard $(MAIN_PACKAGE) ; \
+  done
+
+.PHONY: prod
+prod: build
+	$(PROD_BINARY) --kubeconfig=$(KUBECONFIG) \
+		--sidecar-host=$(SIDECAR_HOST) \
+		--auto-generate-certificates \
+		--locale-config=dist/amd64/locale_conf.json \
+		--bind-address=127.0.0.1 \
+		--port=8080
+
+.PHONY: test-backend
+test-backend: ensure-go
+	go test $(MAIN_PACKAGE)/...
+
+.PHONY: test-frontend
+test-frontend:
+	npm run test
```

Comment:
This results in infinite loop if we run `npm run test` from prompt.

Reply:
`npx jest -c aio/jest.config.js`, right?

Reply:
It was supposed to be `npm run test:frontend` but then I switched something. Thanks!